build/source/engine/f2008funcs.f90

```diff
@@ -119,51 +111,18 @@ subroutine cloneStruc_iv(dataVec,source,mold,err,message)
   err=20; return
  endif
 
- ! get the bounds of the source or the mold vector
- if(present(source)) call getVecBounds(source,lowerBound,upperBound,err,cmessage)
- if(present(mold))   call getVecBounds(mold,  lowerBound,upperBound,err,cmessage)
- if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
+ ! get the upper bounds of the source or the mold vector
+ if(present(source))then; upperBound=ubound(source); endif
+ if(present(mold))  then; upperBound=ubound(source); endif
```

Comment:
Yes, fixed.

These are overloaded subroutines (one for real, one for integer). The code is identical apart from the data type. This seems very cumbersome, though it simply seems to be the way things are done.

It is possible to have a more elegant solution using generic type bound procedures. I'm reluctant to go this route, since this code will ultimately be replaced with the intrinsic `allocate(a, source=b)`